src/main/java/org/opensearch/neuralsearch/util/NeuralSearchClusterUtil.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.util;
+
+import java.util.Locale;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.Version;
+import org.opensearch.cluster.service.ClusterService;
+
+/**
+ * Class abstracts information related to underlying OpenSearch cluster
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@Log4j2
+public class NeuralSearchClusterUtil {
+    private ClusterService clusterService;
+
+    private static NeuralSearchClusterUtil instance;
+
+    /**
+     * Return instance of the cluster context, must be initialized first for proper usage
+     * @return instance of cluster context
+     */
+    public static synchronized NeuralSearchClusterUtil instance() {
+        if (instance == null) {
+            instance = new NeuralSearchClusterUtil();
+        }
+        return instance;
+    }
+
+    /**
+     * Initializes instance of cluster context by injecting dependencies
+     * @param clusterService
+     */
+    public void initialize(final ClusterService clusterService) {
+        this.clusterService = clusterService;
+    }
+
+    /**
+     * Return minimal OpenSearch version based on all nodes currently discoverable in the cluster
+     * @return minimal installed OpenSearch version, default to Version.CURRENT which is typically the latest version
+     */
+    public Version getClusterMinVersion() {
+        try {
+            return this.clusterService.state().getNodes().getMinNodeVersion();
+        } catch (Exception exception) {
+            log.error(
```

Comment:
This seems dangerous to suppress all exceptions. This looks like it was taken from https://github.com/opensearch-project/k-NN/blob/ca5e483e1e70abdc19196e1018b3a7fd06908bf6/src/main/java/org/opensearch/knn/index/KNNClusterUtil.java#L46-L56. @martin-gaievski What was reasoning behind doing this?

Reply:
if I'm not mistaken, the idea was to allow cluster work with a `current` version instead of failing, maybe that core api was throwing some exception at that time. I checked core code, seems it's pretty straightforward now, I think we can get rid of that catch_all approach

Reply:
Taken care off now